src/openms/source/TRANSFORMATIONS/FEATUREFINDER/MultiplexDeltaMassesGenerator.cpp

```diff
@@ -224,86 +249,112 @@ namespace OpenMS
     else if (n == 2)
     {
       // add singlets
-      std::vector<double> singlet(1, 0);
-      mass_pattern_list_.push_back(MultiplexDeltaMasses(singlet));
+      MultiplexDeltaMasses dm;
+      dm.addDeltaMass(MultiplexDeltaMasses::DeltaMass(0,"any_label"));    // There are two singlets with different label sets. But only a single singlet with "any_label" is added.
+      mass_pattern_list_.push_back(dm);
     }
     else if (n == 3)
     {
       for (unsigned i = 0; i < m; ++i)
       {
         // add doublets
-        std::vector<double> doublet1(1, 0);
-        doublet1.push_back(mass_pattern_list_[i].getMassShiftAt(1) - mass_pattern_list_[i].getMassShiftAt(0));
-        mass_pattern_list_.push_back(MultiplexDeltaMasses(doublet1));
-
-        std::vector<double> doublet2(1, 0);
-        doublet2.push_back(mass_pattern_list_[i].getMassShiftAt(2) - mass_pattern_list_[i].getMassShiftAt(1));
-        mass_pattern_list_.push_back(MultiplexDeltaMasses(doublet2));
-
-        std::vector<double> doublet3(1, 0);
-        doublet3.push_back(mass_pattern_list_[i].getMassShiftAt(2) - mass_pattern_list_[i].getMassShiftAt(0));
-        mass_pattern_list_.push_back(MultiplexDeltaMasses(doublet3));
+        MultiplexDeltaMasses doublet1;
+        doublet1.addDeltaMass(mass_pattern_list_[i].getDeltaMasses()[0]);
```

Comment:
`doublet1.getDeltaMasses().push_back(mass_pattern_list_[i].getDeltaMasses()[0]))`
@timosachsenberg Don't you think for readability sake, there is a case for `addDeltaMass()`?